lib/pry/input_completer.rb

```diff
@@ -239,4 +227,30 @@ def build_path(input)
     end
     return path, input
   end
+  
+    private
+
+    def self.fetch_obj_space_collection
+      (self.obj_spase_mutex ||= Mutex.new).synchronize do
+        count = ObjectSpace.each_object(Module).count
```

Comment:
i meant why not have a `collection` local variable on this line, and then use that on L239? i don't think see why you would need to recalculate anything when the scope is within the same method.